ownCloud/Client/Actions/Actions+Extensions/DeleteAction.swift

```diff
@@ -28,19 +28,12 @@ class DeleteAction : Action {
 
 	// MARK: - Extension matching
 	override class func applicablePosition(forContext: ActionContext) -> ActionPosition {
-		let sharedWithUser = forContext.items.sharedWithUser
-
-		if let core = forContext.core {
-			for sharedItem in sharedWithUser {
-				if sharedItem.isShareRootItem(from: core) {
-					return .none
-				}
-			}
+		guard forContext.items.sharedWithUser.count == 0 else {
+			return .none
```

Comment:
Unfortunately, this check is too simple:
- the shared item itself can only be unshared, not deleted
- any item inside a shared folder may itself be deletable

Simplying the check in this way removes the option to delete a file for all shared items, including those inside a shared folder.

Reply:
@felix-schwarz please have a look at commit c54e9dc. Tried to bring back required functionality and at the same time to avoid calling `OCItem.isShareRoot()` which spends a lot of time retrieving parent item from the database using `OCCore.retrieveItemFromDatabase()`.